src/containerapp/azext_containerapp/custom.py

```diff
@@ -1071,7 +1075,7 @@ def create_or_update_github_action(cmd,
                                    registry_url=None,
                                    registry_username=None,
                                    registry_password=None,
-                                   branch="main",
+                                   branch=None,
```

Comment:
Will the change of this default value cause breaking change to users?

Reply:
The change we made here was to default the branch to the GitHub repository's default branch -- so if a user was leaving `--branch` empty *and* their default branch is `main`, there will not be a breaking change for them. Thus I don't think many users will run into a problem with this. Just in case, I will go ahead and add "BREAKING CHANGE" to the history.rst entry